src/state/sagas/solution/FindSolutionById/FindSolutionByIdData.js

```diff
@@ -7,24 +7,41 @@ import { STATUSES } from '../../../commons/Constants';
 import { Api } from '../../../../services/config/Api';
 import { ConfigUtils, SolutionsUtils } from '../../../../utils';
 import { SolutionSchema } from '../../../../services/config/SolutionSchema';
+import { t } from 'i18next';
+import { dispatchSetApplicationErrorMessage } from '../../../dispatchers/app/ApplicationDispatcher';
 
 export function* fetchSolutionByIdData(organizationId, solutionId) {
-  const { data } = yield call(Api.Solutions.findSolutionById, organizationId, solutionId);
-  SolutionsUtils.castMinMaxDefaultValuesInSolution(data);
-  SolutionsUtils.patchSolutionIfLocalConfigExists(data);
-  ConfigUtils.checkUnknownKeysInConfig(SolutionSchema, data);
-  SolutionsUtils.checkParametersValidationConstraintsInSolution(data);
-  ConfigUtils.checkDeprecatedKeysInConfig(data);
-  SolutionsUtils.patchIncompatibleValuesInSolution(data);
+  try {
+    const { data } = yield call(Api.Solutions.findSolutionById, organizationId, solutionId);
+    SolutionsUtils.castMinMaxDefaultValuesInSolution(data);
+    SolutionsUtils.patchSolutionIfLocalConfigExists(data);
+    ConfigUtils.checkUnknownKeysInConfig(SolutionSchema, data);
+    SolutionsUtils.checkParametersValidationConstraintsInSolution(data);
+    ConfigUtils.checkDeprecatedKeysInConfig(data);
+    SolutionsUtils.patchIncompatibleValuesInSolution(data);
 
-  SolutionsUtils.addRunTemplatesParametersIdsDict(data);
-  SolutionsUtils.addTranslationLabels(data);
+    SolutionsUtils.addRunTemplatesParametersIdsDict(data);
+    SolutionsUtils.addTranslationLabels(data);
 
-  yield put({
-    type: SOLUTION_ACTIONS_KEY.SET_CURRENT_SOLUTION,
-    status: STATUSES.SUCCESS,
-    solution: data,
-  });
+    yield put({
+      type: SOLUTION_ACTIONS_KEY.SET_CURRENT_SOLUTION,
+      status: STATUSES.SUCCESS,
+      solution: data,
+    });
+  } catch (error) {
+    console.error(error);
+    yield put(
+      dispatchSetApplicationErrorMessage(
+        error,
+        t('commoncomponents.banner.', "You don't have permission to access this solution.")
```

Comment:
Lack of permissions is probably not the only error that can be caught here (for example, if the workspace configuration has a typo in the solution id, the webapp may try to fetch a solution that doesn't exist).
Can we have a simple distinction between non-existent solutions and permissions errors (based on the return code, for instance), with a default error for other cases: `something went wrong when fetching the solution "<solution id>"`

Reply:
It was a temporary empty key, while fixing the main problem.